sidecar/daemon.go

```diff
@@ -19,13 +20,24 @@ type Daemon struct {
 	publicURL        string
 	ssvClient        api.Ssv
 	server           *http.Server
-	dkg              *dkg.DKGCoordinator
+	dkg              DKGProtocol
 	key              crypto.Keypair
 	thresholdScheme  crypto.ThresholdScheme
 	encryptionScheme crypto.EncryptionScheme
 }
 
+type DKGProtocol interface {
+	RunDKG(identities []crypto.Identity, sessionID []byte, keypair crypto.Keypair) (*kyber.Result, error)
```

Comment:
I feel like we might want to stay away from concrete Kyber types in this repo, so maybe use an interface instead of `kyber.Result` in order to allow us to plug another DKG easily in the future if needed?

Reply:
I've taken the kyber types out of the sidecar now. The downside is we end up juggling back and forth between `[]byte` when we use the scheme, but it's not a big deal